PVTPackage/source/tests/testPublicApi.cpp

```diff
@@ -0,0 +1,279 @@
+                                    convert( apiBuildParams.eosTypes ),
+                                    apiBuildParams.componentNames,
+                                    apiBuildParams.componentMolarWeights,
+                                    apiBuildParams.componentCriticalTemperatures,
+                                    apiBuildParams.componentCriticalPressures,
+                                    apiBuildParams.componentOmegas
+                                  ) );
+      compositionalSystems.push_back( std::move( pair ) );
+      return compositionalSystems.back().second.get();
+    }
+  }
+};
+
+static Holder holder;
+
+void validatePublicApi( const std::string & json_string )
+{
+  const json j = json::parse( json_string );
+
+  const auto refMsp = j.at( PublicAPIKeys::OUTPUT ).get< pds::PDSMSP >();
+
+  const json & computation = j.at( PublicAPIKeys::INPUT ).at( PublicAPIKeys::COMPUTATION );
+  const double pressure = computation.at( PublicAPIKeys::Computation::PRESSURE ).get< double >();
+  const double temperature = computation.at( PublicAPIKeys::Computation::TEMPERATURE ).get< double >();
+  const std::vector< double > feed = computation.at( PublicAPIKeys::Computation::FEED ).get< std::vector< double > >();
+
+  pds::FLASH_TYPE flashType = j.at( PublicAPIKeys::INPUT ).at( PublicAPIKeys::API ).at( FlashKeys::TYPE ).get< pds::FLASH_TYPE >();
+
+
+  pvt::MultiphaseSystem * multiphaseSystem = nullptr;
+  if( flashType == pds::FLASH_TYPE::NEGATIVE_TWO_PHASE or
+      flashType == pds::FLASH_TYPE::FREE_WATER or
+      flashType == pds::FLASH_TYPE::TRIVIAL )
+  {
+    pds::CompositionalApiInputs const apiInputs = j.at( PublicAPIKeys::INPUT ).at( PublicAPIKeys::API ).get< pds::CompositionalApiInputs >();
+    multiphaseSystem = holder.getCompositionalSystem( apiInputs );
+  }
+  if( flashType == pds::FLASH_TYPE::BLACK_OIL or flashType == pds::FLASH_TYPE::DEAD_OIL )
+  {
+    pds::BlackOilDeadOilApiInputs const apiInputs = j.at( PublicAPIKeys::INPUT ).at( PublicAPIKeys::API ).get< pds::BlackOilDeadOilApiInputs >();
+    multiphaseSystem = holder.getDeadOilBlackOilSystem( flashType, apiInputs );
+  }
+
+  multiphaseSystem->Update( pressure, temperature, feed );
+  pvt::MultiphaseSystemProperties const & actualMsp = multiphaseSystem->getMultiphaseSystemProperties();
+
+  compare( actualMsp, refMsp );
+
+  // Building new json reference, uncomment lines below to display computed values as json.
+//  auto newRef = json{
+//    { PublicAPIKeys::INPUT,  j.at( PublicAPIKeys::INPUT )},
+//    { PublicAPIKeys::OUTPUT, actualMsp }
+//  };
+//  std::cout << newRef << std::endl;
+}
+
+TEST( pvt, publicApi )
+{
+  // FIXME Is there a simple way to use data providers with gtest?
+  const std::vector< std::string > fileNames{
+    "pvt_data.txt",
+    "4comp_2ph_1d.log.err",
+    "dead_oil_wells_2d.log.err",
+    "compositional_multiphase_wells_1d.log.err",
+    "compositional_multiphase_wells_2d.log.err",
```

Comment:
I could not find these files in the PR. Do you grab them from the `integratedTests` repository or do you have a local copy?

Reply:
I removed the files from the list. They were my files but they are too big to commit (there is some hammer/fly flavour) here.
I can send them to you if you want.